### PR TITLE
Moonlight vigil

### PR DESCRIPTION
**Pull request by Victor-C-Zhang**
A command-line script to test converters at designated times throughout the day.
- Makes sure your logic is sound for all times of day and night
- Will alert you if your converter encounters unexpected input or is unable to pull the feed data
- Cross-check expected vs actual feed data and conversion results with a dedicated output folder

Currently supports BART only. Subsequent PRs will generalize once general APIs are established.

### CODE REVIEW

File: moonlight_vigil/main.cpp [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  while (fgets(buffer.data(), buffer.size(), pipe.get()) != nullptr) {
+    result += buffer.data();
+  }
+  return result;
+}
+
+/**
+ * @brief Schedule a cron-like job to run web queries and test your converter throughout the day
+ *        or night. Takes 2 parameters:
+ *          - a FULLY QUALIFIED path ts_file to a list of unix timestamps to run your converter on.
```
Comment (q349wang) [inline moonlight_vigil/main.cpp]: No relative paths?

File: moonlight_vigil/main.cpp [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  }
+  ts_file.close();
+  std::sort(ts.begin(), ts.end());
+  auto now = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now());
+  auto it = std::upper_bound(ts.begin(), ts.end(), now);
+  if (it != ts.begin()) {
+    ts.erase(ts.begin(), it);
+  }
+  std::cout << "Scheduled: [" << std::endl;
+  for (auto unix_ts : ts) {
```
Comment (q349wang) [inline moonlight_vigil/main.cpp:66]: const auto&

File: moonlight_vigil/main.cpp [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    std::cout << std::ctime(&unix_ts);
+  }
+  std::cout << "]" << std::endl;
+  std::cout << "Press Ctrl+C to terminate at any time." << std::endl;
+
+  for (auto unix_ts : ts) {
```
Comment (q349wang) [inline moonlight_vigil/main.cpp:72]: Same

**Review by q349wang — APPROVED** [review]
Is this intended to be a one shot executable not a cronjob esque thing? I e. every time you run it you need to update the timestamps to specific times.

File: moonlight_vigil/main.cpp (hunk 2 above)
Comment (Victor-C-Zhang) [reply moonlight_vigil/main.cpp:66]: these are primitives. I refuse to use references for primitives

**Comment by Victor-C-Zhang** [comment]
Yeah, it's supposed to be a one-shot thing. Dunno how protective transit agencies are over their APIs

File: moonlight_vigil/main.cpp (hunk 1 above)
Comment (Victor-C-Zhang) [reply moonlight_vigil/main.cpp]: i guess u can do relative too

**Pull request merged**